filament/src/materials/skybox.mat

```diff
@@ -29,26 +29,108 @@ material {
+            -0.0041960863 * l - 0.7034186147 * m + 1.7076147010 * s
+        );
+    }
+
+    float computeGradient(float t) {
+        const float DIMMING_FACTOR = 0.6;
+        const float GRAD_START_HIGH = 0.9;
+        const float GRAD_LOCATION = 0.4;
+        const float GRAD_START_LOW = 0.2;
+
+        float grad;
+        if (t >= GRAD_LOCATION) {
+            grad = DIMMING_FACTOR + (t - GRAD_LOCATION) * (1.0 / (GRAD_START_HIGH - GRAD_LOCATION)) * (1.0 - DIMMING_FACTOR);
+        } else {
+            grad = 1.0 - (t - GRAD_START_LOW) * (1.0 / (GRAD_LOCATION - GRAD_START_LOW)) * (1.0 - DIMMING_FACTOR);
+        }
+        return saturate(grad);
+    }
+
+    float3 shaprSkyGradient() {
+        const float SHAPR_THRESH = 0.5;
+
+        float upDirectionMapped = normalize(variable_eyeDirection).y * 0.5 + 0.5;
+        float3 colorRGB = approxInverseSRGB(materialParams.color.rgb);
+        float3 colorOklab = linear_srgb_to_oklab(colorRGB);
+        if (colorOklab.r > SHAPR_THRESH) {
+            // oklab.r is the luminance channel
+            colorOklab.r *= computeGradient(upDirectionMapped);
+        } else {
+            const float SHAPR_DARK_COLORS_BIAS = 2.0;
+            // oklab.r is the luminance channel
+            colorOklab.r *= (SHAPR_DARK_COLORS_BIAS - computeGradient(upDirectionMapped));
+        }
+        return oklab_to_linear_srgb(colorOklab);
+    }
+
     void material(inout MaterialInputs material) {
         prepareMaterial(material);
-        vec4 sky;
-        if (materialParams.constantColor != 0) {
-            sky = materialParams.color;
+        float4 sky;
+        if (materialParams.skyboxType == 0) {
+            sky = float4(approxInverseSRGB(materialParams.color.rgb), materialParams.color.a);
+        } else if (materialParams.skyboxType == 1) {
+            float3 shaprSkyColor = shaprSkyGradient();
+            sky = float4(shaprSkyColor, 1.0);
         } else {
-            sky = vec4(textureLod(materialParams_skybox, variable_eyeDirection.xyz, 0.0).rgb, 1.0);
+            sky = float4(textureLod(materialParams_skybox, variable_eyeDirection.xyz, 0.0).rgb, 1.0);
             sky.rgb *= frameUniforms.iblLuminance;
         }
+
         if (materialParams.showSun != 0 && frameUniforms.sun.w >= 0.0f) {
-            vec3 direction = normalize(variable_eyeDirection.xyz);
+            float3 direction = normalize(variable_eyeDirection.xyz);
             // Assume the sun is a sphere
-            vec3 sun = frameUniforms.lightColorIntensity.rgb *
+            float3 sun = frameUniforms.lightColorIntensity.rgb *
                     (frameUniforms.lightColorIntensity.a * (4.0 * PI));
             float cosAngle = dot(direction, frameUniforms.lightDirection);
             float x = (cosAngle - frameUniforms.sun.x) * frameUniforms.sun.z;
             float gradient = pow(1.0 - saturate(x), frameUniforms.sun.w);
             sky.rgb = sky.rgb + gradient * sun;
```

Comment:
I don't think that we should be rendering the Sun into the procedural backgrounds, as it looks weird to me:
![image](https://user-images.githubusercontent.com/90197216/144423552-2dc23b12-908d-4e6c-acf1-33219ddc757e.png)
So let's either turn `showSun` off for the procedurals or skip Sun disk rendering for the non-skybox `skyboxType`s.

Reply:
Fixed in https://github.com/shapr3d/filament/pull/8/commits/91d5e221aec0759129b6da5ad786383348ffd4a2